tutorials/haskell/haskell-time-tutorial/tutorial.md

```diff
@@ -0,0 +1,523 @@
+---
+title: Time, patterns and a little more
+published: 2021-12-02
+tags: haskell, pattern-synonyms, property-testing
+libraries: time
+language: haskell
+author: Felix Miño
+author-name: Felix Miño
+github-profile: felixminom
+description: In this tutorial we’re going to explore the use of pattern synonyms in Haskell, later we’ll give a brief overview to the Stack Builders’ contribution to the `time` Haskell library and finally we’ll review how to use the utilities that were introduced in the contribution.
+---
+
+As Stack Builders one of the core values of our company is contributing to open
+source (OSS). We believe that we have the power to change people’s lives (or at
+least make it easier) by pushing the boundaries of the software industry and
+we’re happy to announce that we were able to contribute to one of Haskell's core
+libraries, [time][time-hackage].
+
+All started with the idea: “What if we bring some utilities that Ruby on Rails
+(RoR) has to Haskell?”. For instance, “How can one get the range of days that a
+month has?” or “How can one get a quarter’s boundaries?” were some of the
+questions that came to our mind. In the following example the RoR code needed to
+get all days in a month will be shown and then Haskell version to do so:
+
+```ruby
+today = Date.today
+today.next_month.all_month
+=> Wed, 01 Nov 2021..Fri, 30 Nov 2021
+```
+
+```haskell
+allMonth :: Day -> [Day]
+allMonth day =
+    let (y, m , _) = toGregorian day
+    in [fromGregorian y m 1 .. fromGregorian y m 31]
+```
+
+and then in `ghci` this function can be invoked:
+
+```bash
+> today <- utctDay <$> getCurrentTime
+> allMonth today
+[2021-11-01 .. 2021-11-30]
+```
+
+As you can see, the Haskell code has to be implemented to latter be called. Why
```

Comment:
s/latter/then (but perhaps rephrasing should be better, just something other than _latter_, which is a typo)